components/atoms/TopUserCard/top-user-card.tsx

```diff
@@ -1,31 +1,77 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
+
+import Link from "next/link";
+import { useRouter } from "next/router";
 import { getAvatarByUsername } from "lib/utils/github";
+import useFollowUser from "lib/hooks/useFollowUser";
+import useSupabaseAuth from "lib/hooks/useSupabaseAuth";
+
 import Avatar from "../Avatar/avatar";
 import Button from "../Button/button";
 
 export interface TopUserCardProps {
-  clasName?: string;
-  following: boolean;
-  username: string;
+  login: string;
+  refreshCallback?: () => void;
 }
-/**
- *
- * additional props to handle follow button click will be added in future versions
- */
 
-const TopUserCard = ({ username, following }: TopUserCardProps) => {
+const TopUserCard = ({ login, refreshCallback }: TopUserCardProps) => {
+  const router = useRouter();
+  const currentPath = router.asPath;
+
+  const { isError: notFollowing, follow, unFollow } = useFollowUser(login);
+  const [host, setHost] = useState("");
+  const { sessionToken, signIn } = useSupabaseAuth();
+
+  const handleFollowUser = async () => {
+    try {
+      if (notFollowing) {
+        await follow();
+        return;
+      }
+      await unFollow();
+    } catch (error) {
+      console.log(error);
+    } finally {
+      refreshCallback && refreshCallback();
```

Comment:
IMHO this ..refreshCall must be [HERE](https://github.com/open-sauced/insights/blob/5a09b62ee96a9f50ca8c6dde3999bf32f7ae072b/lib/hooks/useFollowUser.ts#L16) and you can optimize to make only one function, `follow` and pass an argument for the method of req.